### PR TITLE
Fix unsafe cast in RowExpressionFormatter

### PR DESCRIPTION
**Pull request by lightseba**
Fixed the ClassCastException that occurs when the RowExpression containing a compiled LIKE predicate (RowExpressionInterpreter.tryHandleLike()) is printed with RowExpressionFormatter.

Test plan: unit tests
```
== RELEASE NOTES ==

General Changes
* Fixed a casting error that sometimes occurred for EXPLAIN queries when the query plan contains LIKE predicates
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/RowExpressionFormatter.java [hunk 1]
```diff
@@ -85,20 +85,25 @@ else if (standardFunctionResolution.isBetweenFunction(node.getFunctionHandle()))
...
+                if (patternRowExpression instanceof CallExpression) {
+                    CallExpression patternCallExpression = (CallExpression) patternRowExpression;
+                    // second LIKE argument is:
+                    //  CAST(pattern as LikePattern), if escape is not present
+                    //  LIKE_PATTERN(pattern, escape), if escape is present
+                    if (standardFunctionResolution.isCastFunction(patternCallExpression.getFunctionHandle())) {
+                        RowExpression pattern = patternCallExpression.getArguments().get(0);
+                        return String.format("%s LIKE %s", formatRowExpression(session, value), formatRowExpression(session, pattern));
+                    }
+                    else if (standardFunctionResolution.isLikePatternFunction(patternCallExpression.getFunctionHandle())) {
```
Comment (kaikalur) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/RowExpressionFormatter.java:99]: Does this belong inside the if? Looks like it's unrelated

**Review by kaikalur — CHANGES_REQUESTED** [review]
We need a test that can reproduce the issue.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/RowExpressionFormatter.java (hunk 1 above)
Comment (lightseba) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/RowExpressionFormatter.java:99]: Yes, those are created [here](https://github.com/prestodb/presto/blob/79d8a59d5a42baeb41ac3f21c3c9c7323782e30d/presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java#L735-L757).

**Comment by lightseba** [comment]
> We need a test that can reproduce the issue.

@kaikalur added tests that reproduce the issue

**Review by kaikalur — APPROVED** [review]
LGTM

**Comment by rschlussel** [comment]
Can you make sure your commit message follows the guidelines here: https://chris.beams.io/posts/git-commit/.  Also, add a description to the commit message body about what was fixed (what previously failed and what error we would see).

Also, this should have a release note about what was fixed.

**Comment by lightseba** [comment]
@rschlussel added more info in the commit message + added release note

**Comment by rschlussel** [comment]
Thanks! can you also change the commit message title from "Fixed..." to "Fix..." as per the commit guidelines

**Review by shixuan-fan — APPROVED** [review]
Can you also fix the PR title to be consistent with commit title?

**Comment by shixuan-fan** [comment]
Merged since PR title could be fixed afterwards :)
